### ISSUE TITLE
Medical Facility or Medical Vehicle not working if vehicle is BLUEFOR class

### ISSUE THREAD

**Issue by tourist79** [acemod/ACE3#2423]
ACE3 Version: 3.2.1 stable

**Mods:**
* `@CBA_A3`
* `@ace`

**Placed ACE3 Modules:**

medical settings
advanced medical settings
revive settings 
set medic class
set medical facility
set medical vehicle 
advanced ballistics 
wind deflection 
hearing 
explosives 
interaction 
pointing
map 
DAGR 
name tags 

**Description:**

According to the guidelines given in ACE3 Documentation, I set the above listed modules and within the medical modules I set that surgical kits can be used anywhere, but Personal Aid Kits only at the medical facility and medical vehicle.   Both can be used only by medics.  

When synching the medical facility module with a Huron Container or any EMPTY class ground or air  vehicle, these objects give the option to use the PAK, if the player is a medic per set medic class module.  Elsewhere, only the surgical kit is available for use by the medic player.  

However, when synching the medical facility module with any BLUEFOR class ground or air  vehicle that has a crew, these vehicles DO NOT give the option to use the PAK! 

But the source of the problem is not the fact that the vehicles have a crew, but that they belong to BLUEFOR.  For proof, see my linked repro mission: an empty Chinook which gets a crew by ordering a pilot to the pilot seat and a passenger to cargo via their INIT lines still works as medical vehicle whereas the BLUEFOR Chinook who spawns with his crew doesn't work as medical vehicle)

**Steps to reproduce:**
1) Place Rifle Squad with player as leader  

2) Place an empty chinook and a BLUEFOR/NATO/AIR Chinook (and/or do the same with HEMMT trucks) 

3) Place two crewmembers near the empty Chinook/HEMMT and order them via their INIT to get in as pilot/driver and cargo passenger

4) Place an advanced medical crate 

5) Place all the above listed modules and set them in the way that advanced medical system allows the use of the surgical kit anywhere, but the PAK only at medical facilities and vehicles 

6) Make sure the player is set as medic via listing him in the set medic class module's field 

7) Shoot one of your soldiers full auto to inflict several wounds and make him pass out

8) Grab all you need from the crate, then take care of the wounds, bloodloss, heart rate, stitch him up with surgical kit and give CPR 

9) Carry him to the BLUEFOR class Chinook/HEMMT: PAK usage NOT AVAILABLE 

10) Carry him to the EMPTY class Chinook/HEMMT: PAK usage AVAILABLE

11) If not reproducable: look for possible user errors made by me in creating my repro mission

**Where did the issue occur?**
"Editor" 

**RPT log file:**
http://www.filedropper.com/arma32015-09-1414-56-50  

or 

<a href=http://www.filedropper.com/arma32015-09-1414-56-50><img src=http://www.filedropper.com/download_button.png width=127 height=145 border=0/></a><br /><div style=font-size:9px;font-family:Arial, Helvetica, sans-serif;width:127px;font-color:#44a854;> <a href=http://www.filedropper.com >file storage online</a></div> 

Repro Mission: 

http://www.filedropper.com/ace20medical20teststratis 

or 

<a href=http://www.filedropper.com/ace20medical20teststratis><img src=http://www.filedropper.com/download_button.png width=127 height=145 border=0/></a><br /><div style=font-size:9px;font-family:Arial, Helvetica, sans-serif;width:127px;font-color:#44a854;> <a href=http://www.filedropper.com >file upload storage</a></div>

**Comment by alganthe** [comment]
It's pretty simple really, use the module called "set medical vehicle" instead of  "set medical facility"

**Comment by tourist79** [comment]
NO!  It's not that "simple"!  Granted, I forgot to write that I also have set it up with the medical vehicle module synched with the BLUEFOR Helis and trucks in my repro mission.  It still doesn't work.

**Comment by alganthe** [comment]
In your repro mission you have PAKs set to be used only when the patient is stable, here's your issue.

**Comment by tourist79** [comment]
Nevertheless, alganthe, the PAK IS useable as soon as the patient isn't actively bleeding anymore - I've tried it and CAN USE it on heavily wounded, but bandaged PATIENTS, even on unconscious patients THAT HAVE NO PULSE, as long as I'm JUST NEAR one of the empty class vehicles.  

I don't have to be inside the vehicles - besides, it's a bit inconvenient to fiddle around with the menu when both patient and medic are passsenger in a vehicle, so I'm glad it works outside&close to the vehicle as well.  

So, please, let's look into other possible causes than simple user error - there remains the problem that the entire PAK usage works JUST FINE on one vehicle class but doesn't work on the other vehicle class.

**Comment by Glowbal** [comment]
Hi, thank you for your report.

Using the synchronized tool on a crewed vehicle will result in the crew being classified as, in this case, a medical facility.

The medical facility module is not mean to be used on vehicles, but only on buildings and static objects. I'd advise to switch to the medical vehicle module instead. 

Otherwise, you can use the `setvariable` approach.
```sqf
this setvariable ["ace_medical_isMedicalFacility", true];
```

**Comment by tourist79** [comment]
@Glowbal: just to clarify, should I use the INIT code you posted here for a crewed vehicle instead of the medical vehicle module or in coniunction with it?

**Comment by Glowbal** [comment]
To clarify:

1. The medical facility module is only meant for static objects, such as buildings. It is not intended to be used for vehicles.
2. For vehicles, use the medical vehicle module. Set your treatment requirements accordingly (this is a setting for the PAK). Ensure you use the list approach, not the synchronization tool.
3. If you insist on using a medical facility for a crewed vehicle, you can do so by using the `setvariable` code I gave you an example off.

My recommendation is going with the vehicle module. For details, see the documentation I linked in my previous comment.

**Comment by tourist79** [comment]
Thank you for making things more clear for me! 

I tried both methods and the setvariable code works fine.  Still I have no luck with the medical vehicle module+list approach.  

This is odd, because I use another module+list combo that works: the set medic module with a list of the named AI/player slots that are to be medics is working.   

I tried different ways of naming the vehicles to rule out stuff like lower case/upper case conflict or spacing problems by naming the vehicles medevac1 and medevac2 and by copying that into the list field - doesn't work with that module :-(
